Sources/LanguageServerProtocol/PositionRange.swift

```diff
@@ -45,6 +45,41 @@ public struct PositionRange: CustomCodableWrapper {
   }
 }
 
+/// An LSP-compatible encoding for `Array<Range<Position>>`, for use with `CustomCodable`.
+public struct PositionRangeArray: CustomCodableWrapper {
+  public var wrappedValue: [Range<Position>]
+
+  public init(wrappedValue: [Range<Position>]) {
+    self.wrappedValue = wrappedValue
+  }
+
+  fileprivate enum CodingKeys: String, CodingKey {
+    case lowerBound = "start"
+    case upperBound = "end"
+  }
+
+  public init(from decoder: Decoder) throws {
+    var values: [Range<Position>] = []
+    var arrayContainer = try decoder.unkeyedContainer()
+    while !arrayContainer.isAtEnd {
+      let container = try arrayContainer.nestedContainer(keyedBy: CodingKeys.self)
```

Comment:
Can this be simplified to something like
```swift
let range = try arrayContainer.decode(PositionRange.self)
values.append(range.wrappedValue)
```

Incidentally, I think we could do something like

```swift
values.reserveCapacity(arrayContainer.count ?? 1)
```